Blocks up to Minecraft 1.4

akudeukie opened the pull request:
Added code to handle drawing of new block tiles. Also added handling of cobblestone wall intersections while rendering world tiles.

Following blocks are included:
- Carrot crops;
- Potato crops;
- Cobblestone/Moss stone walls;
- Command block;
- Beacon;
- Anvil;
- Wooden button;
- Flower pots.

Following things are NOT included:
- Head (would require additional prerequisites - mob textures);
- Item frames (they are entities, not blocks?).

Moved deinterpolate() method higher, so it is declared before new functions for drawing new blocks.

Changed how fence gate is being drawn (it's a bit taller now, but still crappy).

slicedlime committed:
Implemented JPEG support with two new command line options:
-f sets the format (png, jpeg or both)
-j sets jpeq quality

slicedlime committed:
Added the possibility of running "make mode=debug to get debuggable binaries.

slicedlime committed:
Added _DEBUG and NDEBUG defines.

slicedlime committed:
- Replaced macros with inline functions for debuggability.
- Optimized former macro functions a fair bit.
- Use a struct with id/data to simplify code a lot.
- Got rid of uninitialized variables by using the above struct as return value.
- Fixed crash bug when trying to shadow the bottom block in a row (access block -1)
- Added a few asserts to sanity check function inputs/outputs.

slicedlime committed:
Fixed all -Wall warnings, added -Wall to default build flags.

slicedlime committed:
Fixed all -Wall warnings, added -Wall to default build flags.

slicedlime committed:
Fixed compile error on some platforms

akudeukie committed:
new blocks in Minecraft 1.4; carrots, potatoes

Added new blocks:
- carrot crops
- potato crops

slicedlime committed:
Fixed bug causing fully zoomed in tiles to be written with bad file extension.

slicedlime committed:
Optimization: Around 35% reduction in rendering time.

slicedlime committed:
Added "coverage" and "profile" compile modes to be able to to some performance analysis

slicedlime committed:
Optimizations that cut away another 15-20% rendering time:
- Simplified some access code that was doing things in over-complex ways or using needless branching.
- Moved common early-outs earlier to avoid doing calculations.
- Delay doing some calculations until really needed.
- Store block data in uint16 internally, instead of doing conversion from source format (mcr/anvil). This takes up 4 bits of wasted space per block but speeds up access. Complexity of access has now been moved to load time, but we access blocks far more often than we load chunks.

slicedlime committed:
Compile fix.

slicedlime committed:
Merge branch 'master' of git://github.com/akudeukie/pigmap

akudeukie committed:
new blocks in Minecraft 1.4; cobblestone walls

Added new blocks:
- Command Block
- Implemented rendering of Cobblestone/Moss walls (acceptable, but with
minor anomalies in rendered tiles in blocks-X.png)

akudeukie committed:
new blocks in Minecraft 1.4;

Added new blocks:
- Anvil;
- Beacon;
- Wooden button.
Improved cobblestone wall sprite generation (final).

akudeukie committed:
new blocks in Minecraft 1.4;

Cleaned up hardcoded values in Beacon drawing code;
Added extra detail to Anvil block.

akudeukie committed:
Flower pots; Reworked base tile drawing code(DRY)

- Added flower pots!
- Made fence gate more prominent (taller);
- Cleaned up and reworked base tile drawing code; Now it draws precise
and correct base tiles; Also made it easier to reuse.

dominickpastore committed:
Removed execute permissions on source files

dominickpastore committed:
Removed trailing carriage returns

dominickpastore committed:
Removed trailing carriage returns and fixed permissions

dominickpastore committed:
Merge branch 'akudeukie-master' into akudeukie-changes

Conflicts:
	Makefile
	blockimages.cpp
	blockimages.h
	chunk.cpp
	chunk.h
	map.cpp
	map.h
	pigmap.cpp
	region.h
	render.cpp
	rgba.cpp
	rgba.h
	tables.cpp
	tables.h
	template.html
	utils.cpp

dominickpastore committed:
Fixed oversight from merge

dominickpastore committed:
Fixed bug causing ice to be drawn improperly sometimes

dominickpastore committed:
Fixed compiler warnings

dantealiegri committed:
fix uint compliation erro

dantealiegri committed:
ignore object and executable

dantealiegri committed:
change threads to -t, make -h the standard help display.

toomuchpete commented:
For the record: I can't get this to run with a custom graphics library. No matter what I try, it tells me it can't find one of the required image files (but not which one).

I have tried copying an entire tileset (with the chest pngs in items/), I've tried moving those pngs into the main folder, I've tried symlinks, no symlinks, etc. Nothing works for me. Haven't actually been able to get this to render anything yet because of that problem.

akudeukie committed:
Blocks from Minecraft 1.5 Snapshots

Major changes:
- Rewrote & refactored block tile generation code;
- Block IDs are no more hard-coded (for the most part);
- Added external block configuration files;
- Added ability to apply few filters to textures (when generating block
tiles);
New blocks from upcoming Minecraft 1.5:
- Trapped Chest;
- Weighted pressure plates (light + heavy);
- Redstone comparator;
- Daylight Sensor;
- Block of Redstone;
- Nether Quartz Ore;
- Hopper;
- Block of Quartz;
- Quartz Stairs;
- Quartz Slab;
- Activator Rail;
- Dropper.
Other changes:
- Added few functions for manipulating RGBAImage;
- Changed to UNIX EOL formatting in modified files.

akudeukie committed:
Make file update

blockimages.cpp requires c++11 standart support, since it uses
<unordered_map>

akudeukie committed:
Merge remote-tracking branch 'dante/master'

Conflicts:
	Makefile
	blockimages.cpp
	blockimages.h
	render.cpp
	rgba.cpp
	rgba.h

Merged with other fork changes (JPEG support, performance improvements, usability improvements)

akudeukie committed:
Post-merge fixes.

version.h file is missing.

akudeukie committed:
Readme update (-t for threads, latest features)

akudeukie committed:
Added check for blockdescriptor file

akudeukie committed:
Change logic for missing/corrupt descriptor file

noidexe commented:
@akudeukie I'm trying to compile your fork but I get the following error:

```
rgba.cpp: In member function ‘bool RGBAImage::writeJPEG(const string&)’:
rgba.cpp:267:29: error: ‘uint’ was not declared in this scope
rgba.cpp:267:34: error: expected ‘;’ before ‘w’
rgba.cpp:267:34: warning: for increment expression has no effect [-Wunused-value]
rgba.cpp:267:35: error: expected ‘)’ before ‘;’ token
rgba.cpp:267:39: error: name lookup of ‘x’ changed for ISO ‘for’ scoping [-fpermissive]
rgba.cpp:267:39: note: (if you use ‘-fpermissive’ G++ will accept your code)
rgba.cpp:267:40: error: expected ‘;’ before ‘)’ token
rgba.cpp:418:1: error: expected ‘}’ at end of input
rgba.cpp:266:14: warning: unused variable ‘p’ [-Wunused-variable]
rgba.cpp:418:1: error: expected ‘}’ at end of input
rgba.cpp:418:1: warning: control reaches end of non-void function [-Wreturn-type]
make: *** [rgba.o] Error 1
```

Using -fpermissive i get:

```
rgba.cpp: In member function ‘bool RGBAImage::writeJPEG(const string&)’:
rgba.cpp:267:29: error: ‘uint’ was not declared in this scope
rgba.cpp:267:34: error: expected ‘;’ before ‘w’
rgba.cpp:267:34: warning: for increment expression has no effect [-Wunused-value]
rgba.cpp:267:35: error: expected ‘)’ before ‘;’ token
rgba.cpp:267:39: warning: name lookup of ‘x’ changed for ISO ‘for’ scoping [-fpermissive]
rgba.cpp:267:17: warning:   using obsolete binding at ‘x’ [-fpermissive]
rgba.cpp:267:40: error: expected ‘;’ before ‘)’ token
rgba.cpp:418:1: error: expected ‘}’ at end of input
rgba.cpp:266:14: warning: unused variable ‘p’ [-Wunused-variable]
rgba.cpp:418:1: error: expected ‘}’ at end of input
rgba.cpp:418:1: warning: control reaches end of non-void function [-Wreturn-type]
make: *** [rgba.o] Error 1
```

The last commit I managed to compile was 3e903d1 . Is there anything I'm doing worng?

My distro is an up-to-date Manjaro Linux(Arch based)

akudeukie commented:
@noidexe

What version of g++ are you using? I was able to compile latest commit on Windows 7 under cygwin environment (g++ 4.5.3), and on Ubuntu Server distribution (g++ 4.7.2).

Also, those compile warnings are result of compiling [`de8c7de`](https://github.com/akudeukie/pigmap/commit/de8c7de666622b078bcd7790838b9517fbde21d4) commit? Because I can't find in code anything, that relates to the provided error messages. Try pulling latest commit into new directory and compiling from scratch (with `make` command).

noidexe commented:
Well, this is embarrasing. It seems that at some point I changed the remote to dominickpastore's and I forgot. I pulled de8c7de and it compiled with no errors.
Thanks for the help! :+1:

Commit:
texture lists are now expected in image directory + fixed some documentation

akudeukie committed:
Merge pull request #1 from UniversE/universe/master

Updated documentation + blocklist location changed (by UniversE)

akudeukie committed:
Blocks from Minecraft 1.6 Snapshots

New blocks from upcoming Minecraft 1.6:
- Hay Block;
- Carpet;
- Hardened Clay;
- Block of Coal.

akudeukie committed:
Blocks from Minecraft 1.6 Snapshots

-Updated up to 13w19a snapshot changes;
-Added Stained Hardened Clay block;
-Aligned wool carpet height to 1 unit/pixel down from 2

Commit:
Merge branch 'master', remote branch 'akudeukie/mc_1.6' into universe/master

Commit:
Merge branches 'universe/master' and 'master'

akudeukie committed:
Hay block fix

- Draw unused hay tiles (data bits) invisible

akudeukie committed:
Merge pull request #2 from akudeukie/mc_1.6

Minecraft 1.6 (1.6.2) - merging back to master branch

akudeukie committed:
Features from Minecraft 1.7

- Aligned all texture names with minecraft 1.6+ resources;
- Fixed comparator rendering (on/off uses same block ID);
- Added new block type for tall flowers (does not affect rendering);
- Added new tile directive - OVERLAY - to combine multiple textures in
one;
- Added special case for rendering tall flowers;
- Added new blocks: packed iced, tall flowers;
- Updated blocks: poppy(ID:38).

akudeukie committed:
Features from Minecraft 1.7

- Added red sand

akudeukie committed:
Alpha filler for RGB images

- Converting RGB to RGBA images (for few textures, that have no alpha
channel; yay! happy times!);
- Removed unnecessary textures.

akudeukie committed:
Features from Minecraft 1.7

- Added stained glass & stained glass pane;
- Glass panes connect with iron bars;
- Cleaned up README.

akudeukie committed:
Features from Minecraft 1.7

- New wood, leaves, planks, stairs, slabs & saplings related to new
trees: Acacia, Dark Oak;
- Some changes to block description types;
- Error output clean-up.

Commit:
Merge branch 'master', remote branch 'akudeukie/mc_1.6'

akudeukie committed:
Merge pull request #3 from akudeukie/mc_1.6

Minecraft 1.7 (1.7.2) features - merging to master

Commit:
fixed double_plant* texture names

akudeukie committed:
Merge pull request #4 from UniversE/master

Texture names are now aligned to 1.7.2

akudeukie committed:
Features from Minecraft 1.8

- Added andesite, granite, diorite (+smooth variants)
- Added slime block

akudeukie committed:
Features from Minecraft 1.8 (14w07a)

- Added iron trapdoor

akudeukie committed:
Features from Minecraft 1.8 (14w25a)

- Added prismarine blocks (rough, bricks, dark variants);
- Added sea lantern;
- Added wet sponge;
- Changed grassless dirt to coarse dirt;
- Fixed farmland (added hydrated farmland).

akudeukie committed:
README update

akudeukie committed:
Features from Minecraft 1.8

- Added new fence variants (acacia, birch, dark oak, jungle, spruce);
- Added new fence gate variants (acacia, birch, dark oak, jungle,
spruce);
- Added inverted daylight sensor;
- Added red sandstone blocks;
- Fixed sandstone/quartz inverted stairs showing wrong texture at the
top of the block;
- New fences and fence gates connect properly.

akudeukie committed:
Features from Minecraft 1.9 (15w31a+)

- Added chorus plant (kinda) and chorus flower;
- Added purpur block, stair and slab;
- Added end stone bricks;

akudeukie committed:
Basic blocks up to recent Minecraft version 16w20a

Blocks from 1.9:
- Added end stone bricks;
- Added beetroot;
- Added grass path;
- Added end gateway;
- Added chained & repeating command blocks; also adjusted standard
command block;
- Added frosted ice;

Blocks from snapshots:
- Added magma block;
- Added nether wart block;
- Added red nether brick;
- Added bone block.

akudeukie committed:
New Blocks up to Minecraft 17w17b snapshot

Blocks from 1.9:
- Added end rod;

Blocks from 1.11:
- Added observer (crude approximation);
- Added shulker boxes;

Blocks from snapshots (1.12):
- Added concrete and concrete powder blocks;
- Added glazed terracotta blocks;
- No colored beds :<

Other notes:
- Fixed EXPAND operator for texture list;
- Fixed import of PNG palette files;
- Archive with textures updated (link in README)
